pageserver/src/http/routes.rs

```diff
@@ -167,10 +169,15 @@ fn build_timeline_info_common(
         }
     };
     let current_physical_size = Some(timeline.layer_size_sum().approximate_is_ok());
-    let state = timeline.current_state();
+    let state = match timeline.current_state() {
+        tenant::TimelineState::Loading => api::TimelineState::Loading,
+        tenant::TimelineState::Active => api::TimelineState::Active,
+        tenant::TimelineState::Stopping => api::TimelineState::Stopping,
+        tenant::TimelineState::Broken => api::TimelineState::Broken,
+    };
```

Comment:
If the objective is to make TimelineState (State) a more local type, we could have

```rust
impl From<tenant::timeline::State> for api::TimelineState {
  fn from(s: tenant::timeline::State) -> Self {
      match s {
        tenant::TimelineState::Loading => api::TimelineState::Loading,
        tenant::TimelineState::Active => api::TimelineState::Active,
        tenant::TimelineState::Stopping => api::TimelineState::Stopping,
        tenant::TimelineState::Broken => api::TimelineState::Broken,
    }
  }
}
```

The above impl could live in pageserver/src/tenant/timeline.rs just fine because I assume this `api` is a dependency of that crate.

And use it here as:

```suggestion
let state = api::TimelineState::from(timeline.current_state());
```

Reply:
Generally I'd agree with this suggestion. But:

My WIP changes that I have on top of this PR remove the `tenant::TimelineState::Broken`.
In those WIP changes, if a timeline is Broken, it's a different type.
The `Tenant::timelines` map then has a value of type

```
enum TimelinesMapEntry {
    Broken(BrokenTimeline),
    NotBroken(Arc<Timeline>),
}
```

(Actually, the plans are bigger, but to make this argument here, the above is sufficient).

The status endpoint then looks more like so:
```
   let state = match timelines_map_entry  {
        NotBroken(tl) => match tl.current_state() {
          tenant::TimelineState::Loading => api::TimelineState::Loading,
          tenant::TimelineState::Active => api::TimelineState::Active,
          tenant::TimelineState::Stopping => api::TimelineState::Stopping,
        }
        Broken(_)  => api::TimelineState::Broken,
    };
```

That match then sits inside a 

```
fn api_state(&self: TimelinesMapEntry) => api::TimelineState {
 ...
}
```

IMO preferable over a `From` impl in this case.